src/utils/country.utils.ts

```diff
@@ -19,14 +20,25 @@ type TraceData = {
  */
 
 export const getCountry = async (): Promise<string> => {
-    try {
-        const response = await fetch(cloudflareTrace).catch(() => null);
-        if (!response) return "";
-        const text = await response.text().catch(() => "");
-        const entries = text ? text.split("\n").map((v) => v.split("=", 2)) : [];
-        const data: TraceData = entries.length ? Object.fromEntries(entries) : {};
-        return data.loc?.toLowerCase() || JSON.parse(Cookies.get("website_status") || "{}")?.loc || "";
-    } catch (error) {
-        return "";
+    if (countryPromise) {
+        return countryPromise;
     }
+
+    countryPromise = (async () => {
+        try {
+            const response = await fetch(cloudflareTrace).catch(() => null);
+            if (!response) {
+                return JSON.parse(Cookies.get("website_status") || "{}")?.clients_country || "";
+            }
+
+            const text = await response.text().catch(() => "");
+            const entries = text ? text.split("\n").map((v) => v.split("=", 2)) : [];
+            const data: TraceData = entries.length ? Object.fromEntries(entries) : {};
+            return data.loc?.toLowerCase() || JSON.parse(Cookies.get("website_status") || "{}")?.clients_country || "";
+        } catch (error) {
+            return JSON.parse(Cookies.get("website_status") || "{}")?.clients_country?.toLowerCase() || "";
```

Comment:
```suggestion
            return JSON.parse(Cookies.get("website_status") || "{}")?.clients_country?.toLowerCase() || "error";
```

the case will keep retried in this case